### PR TITLE
REVAI-4324: Multichannel transcript grouping

### PR DESCRIPTION
**Pull request by dmtrrk**
## Overview
Add new optional parameters to `get_transcript_xx` functions that are available for multichannel media files:
1. `group_channels_by`
Specifies how to group multiple channels in the transcript. This parameter determines the atomic entity for breaking down the transcript into monologues. Only applicable when the submitted media has multiple channels (speaker_channels_count > 1):

- `speaker` - groups by speakers
- `word` - groups by individual words
- `sentence` - groups by complete sentences

2. `group_channels_threshold_ms`
Threshold in milliseconds for handling speaker interruptions. When a speaker interrupts another speaker, this parameter determines how to group the segments. Only applicable when the submitted media has multiple channels (speaker_channels_count > 1):
- If the interruption occurs within this threshold, preference is given to the most recent speaker
- If the interruption occurs after this threshold, a new segment is created

## Usage
```
from rev_ai import apiclient, GroupChannelsType

client = apiclient.RevAiAPIClient(token)
job = client.submit_job_local_file(filePath, speaker_channels_count=2)

# default (word, 1000ms)
transcript = client.get_transcript_text(job.id)

# specific WORD params
transcript_word = client.get_transcript_text(job.id, group_channels_by=GroupChannelsType.WORD, group_channels_threshold_ms=1000)

# specific Sentence parameters
transcript_sentence = client.get_transcript_text(job.id, group_channels_by=GroupChannelsType.SENTENCE, group_channels_threshold_ms=2000)

# Speaker parameter
transcript_speaker = client.get_transcript_text(job.id, group_channels_by=GroupChannelsType.SPEAKER)

```

### CODE REVIEW

File: src/rev_ai/apiclient.py [hunk 1]
```diff
@@ -337,95 +337,144 @@ def get_list_of_jobs(self, limit=None, starting_after=None):
 
         return [Job.from_json(job) for job in response.json()]
 
-    def get_transcript_text(self, id_):
+    def get_transcript_text(self, id_, group_channels_by=None, group_channels_threshold_ms=None):
```
Comment (alexsku) [inline src/rev_ai/apiclient.py:340]: i would use type hints if possible

File: src/rev_ai/apiclient.py [hunk 2]
```diff
@@ -814,3 +863,20 @@ def _create_job_options_payload(
...
+        :param id_: id of job to be requested
+        :param group_channels_by: optional, group channels by speaker or time
+        :param group_channels_threshold_ms: optional, group channels by time threshold, ms
+        :returns: url for getting the transcript
+        """
+        params = []
+        if group_channels_by is not None:
+            params.append('group_channels_by={}'.format(group_channels_by))
+        if group_channels_threshold_ms is not None:
+            params.append('group_channels_threshold_ms={}'.format(group_channels_threshold_ms))
```
Comment (alexsku) [inline src/rev_ai/apiclient.py:891]: @dmtrrk i think you were saying you have doubts about this, i thiunk this is right, we are dealing with these two parameters independently

**Review by alexsku — APPROVED** [review]
:shipit: looks good to me

File: src/rev_ai/apiclient.py (hunk 1 above)
Comment (dmtrrk) [reply src/rev_ai/apiclient.py:340]: I don't see we use type hints in this code. I consider this to be python 2.x compatibility

**Pull request merged**